docs/rules/assertions.md

```diff
@@ -186,6 +196,63 @@ assert/no-pdf-in-ok-response:
   disallowed: ['application/pdf']
 ```
 
+### Custom function example
+
+The following example asserts that `Operation` summary should start with an active verb and have at least three words.
+
+In `.redocly.yaml` describes two custom functions `local/checkWordsStarts` and `local/checkWordsCount`. `local/checkWordsStarts` has in options list of `words`. `local/checkWordsCount` has in options `min` which means that summary field should have `min` words. 
+
+In `plugin.js` inside functions we retrieve this options and make checks. Function called with:
+
+Property | Type | Description
+-- | -- | --
+value | `string` \| [`string`] | Value which selected.
+options | `object` | Options that is described in config file.
+location | `Location Object` | Location in the source document. See [Location Object](../resources/custom-rules.md#location-object)
+
+`.redocly.yaml`
+```yaml
+assert/:
+  subject: Operation
+  property: summary
+  message: Operation summary should start with an active verb
+    local/checkWordsStarts: 
+      words: 
+        - Create
+        - Retrieve
+        - Merge
+        - Delete
+        - List
+        - Upsert
+        - Update
+        - Approve
+        - Reject
+    local/checkWordsCount: 
+      min: 3
+```
+`plugin.js`
+```js
+module.exports = {
+  id: 'local',
+  assertions: {
+    checkWordsStarts: (value, opts, location) => {
+      const regexp = new RegExp(`^${opts.words.join("|")}`)
+      if (regexp.test(value)) {
+        return { isValid: true };
+      }
+      return { isValid: false, location };
```

Comment:
We need to change all our assertions to return a list of problems instead of this object. This will allow a custom function to report more than one problem and we'll be able to provide a default message. (messages are not used right now but we want to make them used as part of assertion finalization)

```suggestion
      if (regexp.test(value)) {
        return []; // empty problems list means assertion passed
      }
      return [{ message: `should start from ${value}`, location };
```

Reply:
Nice idea. We can show message from function if exist, if not -- from assertion.